subgraphs/bridgeworld/src/mappings/randomizer.ts

```diff
@@ -117,11 +117,13 @@ export function handleRandomSeeded(event: RandomSeeded): void {
             ]);
           }
 
-          if (quest.part === 2 && quest.stasisHitCount > 0) {
-            quest.hadStasisPart2 = true;
-          }
-          if (quest.part === 3 && quest.stasisHitCount > 0) {
-            quest.hadStasisPart3 = true;
+          if (quest.stasisHitCount > 0) {
+            if (quest.part === 2) {
+              quest.hadStasisPart2 = true;
+            } else if (quest.part === 3) {
+              quest.hadStasisPart2 = quest.stasisHitCount >= 2;
+              quest.hadStasisPart3 = true;
```

Comment:
Shouldn't these be switched around so part 2 stasis is always true, but part 3 stasis is conditional?

Reply:
Actually never mind this, we will need part 3 to have the stasis for determining auto-progressing correctly on the front-end

Reply:
Coming back to this because I realised this would override `hadStasisPart2` if they just progress from part 2 to 3 and get stasis when they had stasis in part 2 as well

The easiest solution is probably just:
```typescript
quest.hadStasisPart2 = quest.hadStasisPart2 || quest.stasisHitCount >= 2;
```

Reply:
Yep that looks like a good solution, thanks!